### ISSUE TITLE
Datagrid Sortable Columns are not read by Screen Readers

### ISSUE THREAD

**Issue by kalpeshhpatel** [vmware/clarity#4135]
## Describe the bug

For NVDA screen reader, the sortable column in the data-grid is bing read as
`Sortable Column column <column_number>`

## How to reproduce

1) Go to https://v2.clarity.design/datagrid/full
2) Use NVDA screen reader to read the table structure.
3) Observe that columns that are sortable are not recognized by screen-reader.

<img width="1292" alt="Screen Shot 2019-12-19 at 4 09 50 PM" src="https://user-images.githubusercontent.com/5223493/71257740-83979680-235a-11ea-9b6a-89c8e8a13117.png">

## Expected behavior

The screen reader should be able to identify a sortable column name.

For eg. `Name column`, `Creation date column` 

## Versions

This issue is reproducible with the latest Clarity version.

**Comment by gnomeontherun** [comment]
Please provide browser and platform details so we can know how you are testing, some screen readers combinations are better supported than others. Also is this only NVDA on a specific browser?

**Comment by hippee-lee** [comment]
@clane - it seems like the request here is append the projected content text for a column to the generated aria-label on the sort button. The issue seems to be that the screen reader is not reading the projected content for the column. 

 Is there an a11y requirement to be specific with the aria-label for the sort button? e.g should it read `Sort Column Name` or `Sort column Creation Date`? 

Fyi - we generate the aria-label text for the sort button from common strings because that text needs to be i18n-able.

**Comment by kalpeshhpatel** [comment]
> Please provide browser and platform details so we can know how you are testing, some screen readers combinations are better supported than others. Also is this only NVDA on a specific browser?

Hey @gnomeontherun ,

This issue is reproducible with latest Chrome, FF, and Safari too.

Thanks

**Comment by chlane** [comment]
It looks like the tester is saying that rather then hearing the word "column", you should hear the actual column names. The actual column names should be in the aria-label.

**Comment by hippee-lee** [comment]
Ok. Thanks @clane. It looks like we need to explore a more dynamic approach to setting the value for `commonStrings.keys.sortColumn` with common string service.

**Comment by KingMario** [comment]
And I think the sort result should also be read by the screen reader like ascend or descend or not sorted.

**Comment by chlane** [comment]
@KingMario great point. We can use aria-sort for that.  https://www.w3.org/TR/wai-aria-1.1/#aria-sort
